### PR TITLE
Master

### PR DESCRIPTION
**Pull request by mharig**
Pull from hg, bookmarked master.

### CODE REVIEW

File: qb_dl.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+DLCMD = ['wget', '-q', '-c']
+
+
+def qexec(_cmd):
+    with open(environ['QUTE_FIFO'], 'w') as f:
+        f.write(_cmd)
+
+
+def getClipboard():
```
Comment (nimaje) [inline qb_dl.py:36]: why not get the url via argv or QUTE_URL?

File: qb_dl.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    # TODO: progress bar?
+
+
+def download():
+    url = getClipboard()
+    tn = url.split('/')[-1]
+    if not tn:
+        qexec('message-info "URL %s is not a filepath. %s"'%(url, tn))
+        sys.exit()
+    tp = Path(environ['QUTE_DOWNLOAD_DIR']).joinpath(tn)
```
Comment (nimaje) [inline qb_dl.py:63]: why not using --directory-prefix= instead?

**Comment by mharig** [comment]
I want to be able to download a link, without visiting the link, first. 

Strange: I cannot see your pull requests in github.

**Comment by mharig** [comment]
Will investigate that. Thanks.

**Comment by nimaje** [comment]
I only commeted on that commit, I didn't make a pull request.
when using QUTE_URL you could use it in hints, you don't need to visit the url, when using argv to pass the url you don't need to visit the url too.

File: qb_dl.py (hunk 1 above)
Comment (mharig) [reply qb_dl.py:36]: QUTE_URL does not work in any case, e.g. hints do not work when links are displayed by hover functionality.

**Pull request merged**